modules/core/src/main/java/org/apache/ignite/internal/cluster/ClusterGroupAdapter.java

```diff
@@ -805,7 +805,7 @@ private CachesFilter(String cacheName, boolean affNodes, boolean nearNodes, bool
 
     /**
      */
-    private static class AttributeFilter implements IgnitePredicate<ClusterNode> {
+    public static class AttributeFilter implements IgnitePredicate<ClusterNode> {
```

Comment:
I think we can (optionally) set custom node filter in a test and left this class private.